spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/TaskExecutionController.java

```diff
@@ -93,6 +94,18 @@ public TaskExecutionController(TaskExplorer explorer) {
 	}
 
 	/**
+	 * View the details of a single task execution, specified by id.
+	 *
+	 * @param id the id of the requested {@link TaskExecution}
+	 * @return the {@link TaskExecution}
+	 */
+	@RequestMapping(value = "/id/{id}", method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public TaskExecutionResource view(@PathVariable("id") long id) {
+		return taskAssembler.toResource(this.explorer.getTaskExecution(id));
```

Comment:
Throws NPE if task execution id is not present.  Instead should throw as more specific Exception like NoSuchTaskExecutionException.  Here is what  I suggest https://github.com/cppwfs/spring-cloud-dataflow/commit/d10e581bc5c1620b5776b16c8e96a401fb87d639